activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -180,6 +180,10 @@ def spec(config, name = nil)
 
           adapter_method = "#{spec[:adapter]}_connection"
 
+          unless ActiveRecord::Base.respond_to?(adapter_method)
+            raise AdapterNotFound, "database configuration specifies nonexistent #{spec.config[:adapter]} adapter"
```

Comment:
Think this error message is pretty confusing. What should users do if they hit this error?